src/main/java/be/seeseemelk/mockbukkit/block/state/EndGatewayMock.java

```diff
@@ -0,0 +1,81 @@
+package be.seeseemelk.mockbukkit.block.state;
+
+import com.google.common.base.Preconditions;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.EndGateway;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class EndGatewayMock extends TileStateMock implements EndGateway
+{
+
+	private long age;
+	private boolean exactTeleport;
+	private @Nullable Location exitLocation;
+
+	protected EndGatewayMock(@NotNull Material material)
+	{
+		super(material);
+		checkType(material, Material.END_GATEWAY);
+	}
+
+	protected EndGatewayMock(@NotNull Block block)
+	{
+		super(block);
+		checkType(block, Material.END_GATEWAY);
+	}
+
+	protected EndGatewayMock(@NotNull EndGatewayMock state)
+	{
+		super(state);
+		this.age = state.age;
+		this.exactTeleport = state.exactTeleport;
+		this.exitLocation = state.exitLocation;
+	}
+
+	@Override
+	public @NotNull BlockState getSnapshot()
+	{
+		return new EndGatewayMock(this);
+	}
+
+	@Override
+	public @Nullable Location getExitLocation()
+	{
+		return this.exitLocation;
```

Comment:
```suggestion
		return this.exitLocation == null ? null : this.exitLocation.clone();
```
The location shouldn't be modifiable without using `setExitLocation`.